File: src/m/plot/writejsfield.m
function writejsfield(fid,name,variable,nods)
%WRITEJSFIELD - write variable to javascript file 
%
%   Usage:
%      writejsfield(fid,name,variable)
%

	%write array:
	if size(variable,2)==1,
		fprintf(fid,'<!-- %s{{{-->\n',name);
		fprintf(fid,'%s=[',name);
		for i=1:nods-1,
			fprintf(fid,'%g,',variable(i));
		end
		fprintf(fid,'%g];\n',variable(end));
		fprintf(fid,'<!--}}}-->\n');
	else
		%multi-sized array: 
		fprintf(fid,'<!-- %s{{{-->\n',name);
		fprintf(fid,'%s=[]\n',name);
		for i=1:size(variable,2),
			fprintf(fid,'%s["%i"]=[',name,i);
			for j=1:nods-1,
				fprintf(fid,'%g,',variable(j,i));
			end
			fprintf(fid,'%g];\n',variable(end,i));
		end
		fprintf(fid,'<!--}}}-->\n');
	end

File: src/m/plot/writejsfield.py
import numpy as np


def writejsfield(fid, name, variable, nods):
    #WRITEJSFIELD - write variable to javascript file
    #
    #   Usage:
    #      writejsfield(fid, name, variable)
    #
    #write array:
    #if not isinstance(variable, list):
    if type(variable[0]) == np.float64:
        fid.write('<!-- {0}{{{{{{-->\n'.format(name))
        fid.write('{0}=['.format(name))
        for i in range(0, nods - 1):
            fid.write('{0}, '.format(variable[i]))
        fid.write('{0}];\n'.format(variable[-1]))
        fid.write('<!--}}}}}}-->\n')
    else:
        #multi - sized array:
        fid.write('<!-- {0}{{{{{{-->\n'.format(name))
        fid.write('{0} = []\n'.format(name))
        for i in range(0, len(variable[2])):
            fid.write('{0}["{1}"] = ['.format(name, i))
            for j in range(1, nods - 1):
                fid.write('{0}, '.format(variable[j][i]))
            fid.write('{0}];\n'.format(variable[-1][i]))
        fid.write('<!--}}}}}}-->\n')
